src/main/java/com/google/sps/servlets/InfoCardServlet.java

```diff
@@ -59,20 +72,77 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       return;
     }
 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity chosenElection = ServletHelper.findElectionInDatastore(datastore, electionId);
+
+    if (chosenElection == null) {
+      response.setContentType("text/html");
+      response
+          .getWriter()
+          .println(String.format("Could not find election with ID %s in Datastore.", electionId));
+      response.setStatus(400);
+      return;
+    }
+
+    // TODO: if this election is already populated, we don't need to make another query
+
     URL url =
         new URL(
             String.format(
+                "%s?address=%s&electionId=%s&key=%s",
                 BASE_URL,
                 address,
                 electionId,
                 ServletHelper.getApiKey("112408856470", "election-api-key", "1")));
-    JSONObject obj = ServletHelper.readFromApiUrl(url, response);
+    JSONObject electionListFromApi = ServletHelper.readFromApiUrl(url);
+    JSONArray positionListFromApi = electionListFromApi.getJSONArray("contests");
+    chosenElection.setProperty("positions", fillPositions(datastore, positionListFromApi));
```

Comment:
I would pull fillPositions() out to a separate line, so that it's a little more clear that you're actually setting datastore keys pointing to the positions rather than the actual position data itself.